snappy/snap_file.go

```diff
@@ -20,74 +20,29 @@
 package snappy
 
 import (
-	"path/filepath"
-
-	"github.com/ubuntu-core/snappy/dirs"
 	"github.com/ubuntu-core/snappy/snap"
 )
 
-// SnapFile is a local snap file that can get installed
-type SnapFile struct {
-	m   *snapYaml
-	deb snap.File
-
-	instdir string
-}
+// openSnapBlob opens a snap blob returning both a snap.Info completed
+// with sideInfo (if not nil) and a corresponding snap.File.
+func openSnapBlob(snapFile string, unsignedOk bool, sideInfo *snap.SideInfo) (*snap.Info, snap.File, error) {
+	// TODO: what precautions to take if unsignedOk == false ?
 
-// NewSnapFile loads a snap from the given snapFile
-func NewSnapFile(snapFile string, unsignedOk bool) (*SnapFile, error) {
-	d, err := snap.Open(snapFile)
+	blobf, err := snap.Open(snapFile)
```

Comment:
I suggest renaming blobf to snapf everywhere. Commenting here because this seems like the most unclear case, since we're getting a snapFile as a parameter and returning a snap.File. For clarity, I'd rename:
* openSnapBlob => openSnapFile
* snapFile => snapPath
* blobf => snapf (here and everywhere)